### ISSUE TITLE
Force the first element to be a H1

### ISSUE THREAD

**Issue by antoniobrandao** [yabwe/medium-editor#319]
Hello, simple question. Is there a way to force the first element to be a H1 ?

**Comment by csarven** [comment]
If I'm not mistaken the line in question is now at https://github.com/daviferreira/medium-editor/blob/master/src/js/core.js#L403

I'm a bit unclear about this issue. Isn't the purpose of using the core option `firstHeader` for this exact request?

**Comment by antoniobrandao** [comment]
@csarven I thought the same about the purpose of that "firstHeader" option, but that only seemed to affect the first Headline button in the floating Toolbar...

**Comment by nmielnik** [comment]
Yeah, the `firstHeader` option corresponds to the type of header tag to use for the `h1` button in the toolbar.  There is also a `secondHeader` option for what type of header tag to use for the `h2` button in the toolbar.

When I say type of header tag, i mean `h1`, `h2`, `h3`, `h4`, etc.

Seems like this option could have a better name, but that's the purpose of the option today.

**Comment by Fipo** [comment]
Unfortunately I have same issue like @antoniobrandao.
I wan't to use two editor instances - one for Title and one for general post.

**Comment by nicooprat** [comment]
I'm also interested in this feature. Any plan to add it, or any knew workaround?

**Comment by nmielnik** [comment]
This issue was also asked for as #1006 .

So the part of the code that automatically wraps text with a `<p>` has been brought up in a bunch of issues ( #907, #829, #865, #149 ) and is potentially being modified by #997 (thanks @harzzn)

If someone wanted to try adding support for this into the editor, you could try just modifying the `handleKeyUp()` method in [core.js](https://github.com/yabwe/medium-editor/blob/master/src/js/core.js).  In particular, the lines that do this:
```js
this.options.ownerDocument.execCommand('formatBlock', false, 'p');
```
Changing the 'p' passed to `execCommand` to be a tag name that is passed in via an option might be all that is needed, but it would take some experimentation to see how it all behaves and if there are any weird issues that come up.

**Comment by googlicius** [comment]
If you want to treat first heading as title, just use css to makes it bigger, Medium currently also use this on their site:
```css
.content > h3:first-child {
  font-weight: 600;
  font-size: 42px;
  line-height: 1.04;
}
```

**Comment by csarven** [comment]
@googlicius That's a no go for me. The issue is not a stylistic matter. CSS can generally make anything thing *look* different.

**Comment by googlicius** [comment]
Other way is to put h1 tag directly into textarea if content is empty:
```html
<textarea id="editor"><h1><br></h1></textarea>
```
